### ISSUE TITLE
Improve support for placementConstraints and placementStrategy

### ISSUE THREAD

**Issue by efeller** [Shippable/support#4407]
AWS ECS supports placementConstraints (https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-placement-constraints.html). However, shippable seems to only support placementStrategy (http://docs.shippable.com/platform/workflow/resource/dockeroptions/).

On the docs it is mentioned that: "All ECS Service definition parameters documented here can be specified under service since these are passthrough. The same applies for taskDefinition documented here."

However, that does not seem to work. After setting a "placementConstraint" (similarly to placementStrategy) it is not propagated to ECS. Can you please add support for placementConstraints for old and new syntax?

**Comment by efeller** [comment]
Alright I got the placementConstraints working. I had to manually remove the job from ECS and redeploy it via shippable (using reset). Only after that I could see the placementConstraint.

Can this be documented somewhere and/or upgrade of the jobs fixed to take placementConstraints into account without the need to destroy the job?

Thanks

**Comment by trriplejay** [comment]
I can open a feature request to have "placementConstraints" and "placementStrategy" be treated similarly to the "loadBalancers" field, where shippable would detect a change and then automatically delete and recreate your aws service behind the scenes as part of one deployment. This would incur some downtime while the service is drained and recreated, but at least wouldn't require you to delete the job entirely.  Does that sound reasonable? 

Unfortunately this is a limitation of ECS that these fields cannot be updated for an existing service.  Your other option would be to use "blueGreen" mode instead of "upgrade", since then a new service is created for each deployment.

**Comment by efeller** [comment]
Yep, thanks a lot. Sounds very reasonable. I would be fine with that, as in case of a placement constraint change a service downtime is expected anyhow (e.g., ECS instance was terminated and we need to move the service to another instance). In the meantime I will try using "blueGreen" 👍

**Comment by manishas** [comment]
Closing in favor of 
https://github.com/Shippable/support/issues/4406 which will track support for placementStrategy and placementConstraints.
